### PR TITLE
try to fix pipenv

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR tries to fix the Pipenv install steps.

## Fixes

- #831, #832

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -12,6 +12,8 @@ name = "test-pypi"
 bandit = "==1.6.2"
 black = "==19.10b0"
 bs4 = "==0.0.1"
+colorlog = "==4.1.0"
```
Comment (DavidMinarsch) [inline Pipfile]: these are not core dependencies, only needed for tests

File: setup.py [hunk 2]
```diff
@@ -68,15 +68,16 @@ def get_all_extras() -> Dict:
 
     cli_deps = [
         "click",
-        "PyYAML",
-        "jsonschema",
+        "pyyaml>=4.2b1",
```
Comment (DavidMinarsch) [inline setup.py]: we should mirror Pipfile[packages] here imo

File: tox.ini [hunk 3]
```diff
@@ -64,10 +64,10 @@ deps = black
 commands=black aea examples packages scripts tests --check --verbose
 
 [testenv:flake8]
-deps = flake8
-       flake8-docstrings
-       flake8-import-order
-       flake8-bugbear
+deps = flake8==3.7.9
```
Comment (DavidMinarsch) [inline tox.ini]: to mirror Pipfile

File: setup.py (hunk 2 above)
Comment (marcofavorito) [reply setup.py]: not applicable anymore

File: tox.ini [hunk 4]
```diff
@@ -26,98 +26,106 @@ commands =
...
+    pytest-asyncio==0.10.0
+    docker
+    colorlog==4.1.0
+    oef==0.8.1
+    gym==0.15.6
+    numpy==1.18.1
+    tensorflow==1.14.0
 commands =
-    pipenv install
-    pip install -e .[all]
```
Comment (DavidMinarsch) [inline tox.ini]: isn't `pip install -e .[all]` required?

File: tox.ini [hunk 5]
```diff
@@ -26,98 +26,106 @@ commands =
...
+    docker
+    colorlog==4.1.0
+    oef==0.8.1
+    gym==0.15.6
+    numpy==1.18.1
+    tensorflow==1.14.0
 
 commands =
-    pipenv install
-    pip install -e .[all]
```
Comment (DavidMinarsch) [inline tox.ini]: see above

File: tox.ini (hunk 4 above)
Comment (marcofavorito) [reply tox.ini]: it is not required. `tox` provides to install the package by default.
However, a problem I'm encountering is that the coverage is not computed only on the main code base but also on the copy of the codebase `tox` does.

**Pull request merged**